internal/k8s/controllers/http_route_controller_test.go

```diff
@@ -75,3 +86,100 @@ func TestHTTPRoute(t *testing.T) {
 		})
 	}
 }
+
+func TestHTTPRouteReferencePolicyToRouteRequests(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	serviceNamespace := gw.Namespace("namespace3")
+
+	backendObjRef := gw.BackendObjectReference{
+		Name:      gw.ObjectName("service"),
+		Namespace: &serviceNamespace,
+	}
+
+	httpRouteSpec := gw.HTTPRouteSpec{
+		Rules: []gw.HTTPRouteRule{{
+			BackendRefs: []gw.HTTPBackendRef{{
+				BackendRef: gw.BackendRef{
+					BackendObjectReference: backendObjRef,
+				},
+			}},
+		}},
+	}
+
+	refPolicy := gw.ReferencePolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "ReferencePolicy"},
+		ObjectMeta: metav1.ObjectMeta{Namespace: "namespace3"},
+		Spec: gw.ReferencePolicySpec{
+			From: []gw.ReferencePolicyFrom{{
+				Group:     "gateway.networking.k8s.io",
+				Kind:      "HTTPRoute",
+				Namespace: "namespace1",
+			}},
+			To: []gw.ReferencePolicyTo{{
+				Kind: "Service",
+			}},
+		},
+	}
+
+	gatewayclient := NewTestClient(
+		&gw.HTTPRouteList{
+			Items: []gw.HTTPRoute{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "httproute",
+						Namespace: "namespace1",
+					},
+					Spec: httpRouteSpec,
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "httproute",
+						Namespace: "namespace2",
+					},
+					Spec: httpRouteSpec,
+				},
+			},
+		},
+		&refPolicy,
+	)
+
+	controller := &HTTPRouteReconciler{
+		Client:         gatewayclient,
+		Log:            hclog.NewNullLogger(),
+		ControllerName: mockControllerName,
+		Manager:        reconcilerMocks.NewMockReconcileManager(ctrl),
+	}
+
+	requests := controller.referencePolicyToRouteRequests(&refPolicy)
+
+	require.Equal(t, []reconcile.Request{{
+		NamespacedName: types.NamespacedName{
+			Name:      "httproute",
+			Namespace: "namespace1",
+		},
+	}}, requests)
+}
+
+// FIXME: this should be refactored into a test utility package
+func NewTestClient(list client.ObjectList, objects ...client.Object) gatewayclient.Client {
+	scheme := runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(gw.AddToScheme(scheme))
+	apigwv1alpha1.RegisterTypes(scheme)
+
+	builder := fake.
+		NewClientBuilder().
+		WithScheme(scheme)
+	if list != nil {
+		builder = builder.WithLists(list)
+	}
+	if len(objects) > 0 {
+		builder = builder.WithObjects(objects...)
+	}
```

Comment:
`objects` will be an empty list if nothing is passed in, so you don't need a check on `len`
```suggestion
	builder := fake.
		NewClientBuilder().
		WithScheme(scheme).
		WithObjects(objects...)
	if list != nil {
		builder = builder.WithLists(list)
	}
```

Reply:
Implemented in #162 which moves this into a dedicated test helper package.